.github/workflows/formatting.yml

```diff
@@ -0,0 +1,54 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+name: Code Formatting
+on:
+  push:
+    branches-ignore: [ 'dependabot/**' ]
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+jobs:
+  ts-format:
+    name: TypeScript Formatting
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2.3.5
+      - name: Setup Node
+        uses: actions/setup-node@v1.4.6
+        with:
+          node-version: '16'
+      - run: yarn install
+      - run: yarn lint
```

Comment:
How customizable is this lint tool? E.g., can we make it ignore some things that we decide are ok stylistically?

Reply:
Well `yarn lint` is just being removed from the yarn `pre-test` script as linting on a Windows machine causes issues because of line ending differences.

Other than that `prettier` is very customizable with ignoring. If we want to ignore a entire directory, file, type of file, etc you can that to the `.prettierignore` file. Then you can ignore specific lines of file (JavaScript/TypeScript) like this:
```
// prettier-ignore
```

for other file types, eg yaml, they detail that here https://prettier.io/docs/en/ignore.html